src/item.cpp

```diff
@@ -3661,6 +3661,10 @@ bool item::can_contain( const itype &tp ) const
 
 bool item::spill_contents( Character &c )
 {
+    if( is_container_empty() || !is_container() ) {
```

Comment:
It could technically work and be useful for other things.
For example, waterskins too full to fit in inventory without spilling. Destroyed items, which currently drop their contents with hardcoded loops (this one would be quite useful).